src/webhook/test_namespace_validation.py

```diff
@@ -84,33 +90,55 @@ def runtest(self, namespace, groups, expect):
-                groups = gl + ('dedicated-admins',)
-                self.runtest(ns, groups, False)
+        # layered-cs-sre-admins should not have access to any privledged namespaces except ones starting with 'redhat'
+        # dedicated-admins should not have access to any privledged namespaces
+        for ns in self.PRIVILEGED_NAMESPACES:
+            for user in self.NON_PRIVILEGED_USERS:
+                groups = ('layered-cs-sre-admins',)
+                self.runtest(ns, groups, user, False)
+                groups = ('dedicated-admins',)
+                self.runtest(ns, groups, user, False)
+                groups = ('random-test-group',)
+                self.runtest(ns, groups, user, False)
 
     def test_allow_group(self):
-        # If the group list doesn't contain 'dedicated-admins', always ALLOW,
-        # even if the namespace is privileged. (Really?)
-        for ns in PRIVILEGED_NAMESPACES + NONPRIV_NAMESPACES:
-            for gl in GROUP_LISTS:
-                self.runtest(ns, gl, True)
+        # user in dedicated-admins group can edit non-privileged NS
+        # user in osd-sre-admins group can edit non-privileged NS
+        # user in osd-sre-admins group can edit privileged NS
+        for ns in self.PRIVILEGED_NAMESPACES + self.NONPRIV_NAMESPACES + self.REDHAT_NAMESPACES:
+            for user in self.PRIVILEGED_USERS:
+                groups = ('osd-sre-cluster-admins',)
+                self.runtest(ns, groups, user, True)
+                groups = ('osd-sre-admins',)
+                self.runtest(ns, groups, user, True)
+
+    def test_allow_layered_admins(self):
+        # user in layered-sre-cluster-admins group can edit non-privileged NS
+        for ns in self.NONPRIV_NAMESPACES + self.REDHAT_NAMESPACES:
+            for user in self.NON_PRIVILEGED_USERS:
+                groups = ('layered-sre-cluster-admins',)
+                self.runtest(ns, groups, user, True)
+
+    def test_priv_users(self):
+        # all privileged users (kube/system:admin) can edit any privileged or
+        # non-privileged NS
+        for user in self.PRIVILEGED_USERS:
+            for ns in self.PRIVILEGED_NAMESPACES + self.NONPRIV_NAMESPACES + self.REDHAT_NAMESPACES:
+                groups = ('random-test-group',)
+                self.runtest(ns, groups, user, True)
 
     def test_allow_ns(self):
         # Nonprivileged namespaces always ALLOW, even if the group list
         # contains 'dedicated-admins'.
-        for ns in NONPRIV_NAMESPACES:
-            for gl in GROUP_LISTS:
-                self.runtest(ns, gl, True)
-                groups = gl + ('dedicated-admins',)
-                self.runtest(ns, groups, True)
+        for ns in self.NONPRIV_NAMESPACES:
+            for gl in self.GROUP_LISTS:
+                for user in self.NON_PRIVILEGED_USERS:
+                    groups = gl + ('dedicated-admins',)
+                    self.runtest(ns, groups, user, True)
```

Comment:
These test cases are also not providing value, as what you are effectively doing here is ensuring that any user who is a member of both a privileged group and "dedicated-admins" is allowed to edit any non-privileged NS. We already know this should be true, since we tested that membership of these privileged groups earlier. What you want to test here is that anyone who is in a non-privileged group ("random-test-group", "dedicated-admins") can edit any non-privileged NS
```suggestion
        for ns in self.NONPRIV_NAMESPACES:
            for user in self.NON_PRIVILEGED_USERS:
                groups = ('dedicated-admins',)
                self.runtest(ns, groups, user, True)
                groups = ('random-test-group',)
                self.runtest(ns, groups, user, True)
```